gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -84,21 +84,24 @@ private boolean includeField(Field f, boolean serialize) {
   /** first element holds the default name */
   @SuppressWarnings("MixedMutabilityReturnType")
   private List<String> getFieldNames(Field f) {
+
+    String fieldName;
+    List<String> alternates;
     SerializedName annotation = f.getAnnotation(SerializedName.class);
     if (annotation == null) {
-      String name = fieldNamingPolicy.translateName(f);
-      return Collections.singletonList(name);
-    }
-
-    String serializedName = annotation.value();
-    String[] alternates = annotation.alternate();
-    if (alternates.length == 0) {
-      return Collections.singletonList(serializedName);
+      fieldName = fieldNamingPolicy.translateName(f);
+      alternates = fieldNamingPolicy.alternateNames(f);
+    } else {
+      fieldName = annotation.value();
+      alternates = Arrays.asList(annotation.alternate());
+      if (alternates.size() == 0) {
```

Comment:
I think this `if` could be moved outside the containing `if` since it would apply to both cases.